### ISSUE TITLE
File newly added to GH repo will not cache, displays "Package size exceeded" error instead

### ISSUE THREAD

**Issue by adamlui** [jsdelivr/jsdelivr#18571]
### Description

After adding https://github.com/KudoAI/chatgpt.js/blob/main/media/images/logos/chatgpt.js/flat/black/logo600x62.png,
visiting https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/flat/black/logo600x62.png
results in this error:

![image](https://github.com/jsdelivr/jsdelivr/assets/10906554/05090989-86c4-4ee7-8cde-98ef5d399950)

### Affected jsDelivr links

https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/flat/black/logo600x62.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/flat/black/logo5995x619.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/flat/white/logo600x62.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/flat/white/logo5995x619.png

### Response headers

```
HTTP/1.1 403 Forbidden
Connection: keep-alive
Content-Length: 170
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: *
Timing-Allow-Origin: *
Cache-Control: public, max-age=60, s-maxage=60
Cross-Origin-Resource-Policy: cross-origin
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Content-Type: text/plain; charset=utf-8
ETag: W/"aa-oL6vDqvmPuxPwUk2Hk2PqtznjnQ"
Accept-Ranges: bytes
Age: 0
Date: Tue, 07 May 2024 02:42:46 GMT
X-Served-By: cache-fra-etou8220100-FRA, cache-pao-kpao1770069-PAO
X-Cache: MISS, MISS
Vary: Accept-Encoding
alt-svc: h3=":443";ma=86400,h3-29=":443";ma=86400,h3-27=":443";ma=86400
```

### Information

- Device OS: Windows
- Browser & Version: Firefox 125.0.3 (64-bit)
- VPN provider: N/A
- Your location: California, United States

### Requisites

- [X] I performed a [cursory search of the issue tracker](https://github.com/jsdelivr/jsdelivr/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I checked the [documentation](https://github.com/jsdelivr/jsdelivr) to understand that the issue I am reporting is not normal behavior.
- [X] I understand that not filling out this template correctly will lead to the issue being closed.

### Additional content

After performing a cursory search for `is:issue 404 github file`, I found https://github.com/jsdelivr/jsdelivr/issues/18535#issuecomment-1763436495 instructing to insert `@main` into URL (which makes the above 4 links work) but why is it not required for other repo files, for example:

https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/with-reflection/darkmode.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/logos/chatgpt.js/with-reflection/lightmode.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/dist/chatgpt.min.js

**Comment by MartinKolarik** [comment]
Please check now, the files seem to work.

**Comment by adamlui** [comment]
@MartinKolarik yes they do! Did you take manual action? Because it usually goes live within seconds so that was odd

**Comment by MartinKolarik** [comment]
No, I suppose the request to GH failed the first time or something like that.

**Comment by adamlui** [comment]
It's happening again, this time for 

https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/black-on-white/icon189.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/black-on-white/icon50.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-black/icon189.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-black/icon50.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-blue-green/icon189.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-blue-green/icon50.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-gray/icon189.png
https://cdn.jsdelivr.net/gh/KudoAI/chatgpt.js/media/images/icons/platforms/chatgpt/white-on-gray/icon50.png

With the previous 4 PNGs, I had checked probably 6+ hours later and it was still error (without branch tag)

**Comment by adamlui** [comment]
it's only happening with the KudoAI/chatgpt.js repo for some reason, I just added some buttons to adamlui/chatgpt-widescreen and they were instantly served

**Comment by MartinKolarik** [comment]
I see the repo actually has releases: https://github.com/KudoAI/chatgpt.js/tags

If you don't specify a version, it means "latest release" not "latest commit"

**Comment by adamlui** [comment]
I suspected that was a cause since that was the only recent change that stuck out. I personally would expect a tagless GitHub CDN URL to simply point to the latest src files so I could play with newest features because there are repos that sparsely release, and it would seem like the URL for latest GitHub is for that purpose vs. using the npm one.

I think using `@latest` for release + no tag for latest commit would make more sense to users, because not all repos make releases and if users are used to jsDelivr URL behavior for those types, they will be confused when using w/ repos that do release (and vice versa)

**Comment by adamlui** [comment]
...and if they truly want latest release, they could use the /dist instead of /src file, so it's much more flexible
